util/storage-writer/src/csv_storage_writer.rs

```diff
@@ -0,0 +1,84 @@
+// Copyright 2015-2018 Parity Technologies (UK) Ltd.
+// This file is part of Parity.
+
+// Parity is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Parity is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Parity.  If not, see <http://www.gnu.org/licenses/>.
+
+use std::collections::HashMap;
+use std::io;
+use std::fs;
+use std::fs::File;
+use std::sync::{Arc, Mutex};
+
+use super::StorageWriter;
+use csv::Writer;
+use dir::helpers::replace_home;
+use dir::default_data_path;
+use ethereum_types::{Address, H256};
+
+#[derive(Clone)]
+pub struct CsvStorageWriter {
+    watched_contracts: Vec<Address>,
+    writer: Arc<Mutex<Writer<File>>>
+}
+
+impl CsvStorageWriter {
+    pub fn new(watched_contracts: Vec<Address>) -> CsvStorageWriter {
+        let path = replace_home(&default_data_path(), "$BASE/watched_storage");
+
+        // TODO: handle error differently on file creation
```

Comment:
This one I think needs to be finished before we PR upstream since it's literally the only thing in their [coding guide on the wiki](https://wiki.parity.io/Coding-guide) - "don't panic"

Reply:
Welp I'm actually thinking I'm going to leave this how it is since they appear to be doing [something similar](https://github.com/paritytech/parity-ethereum/blob/master/util/journaldb/src/archivedb.rs#L52) when initializing a journaldb. Changing it would require wrapping the `CsvStorageWriter` in a result, which would then bubble up, and I'm curious to see if they have feedback when we PR upstream.